packages/machine/test/unit/cf-operations/set-state.spec.ts

```diff
@@ -0,0 +1,134 @@
+import * as ethers from "ethers";
+
+import * as abi from "../../../src/abi";
+import { CfOpSetState } from "../../../src/middleware/cf-operation/cf-op-setstate";
+import {
+  CfAppInstance,
+  CfAppInterface,
+  Terms
+} from "../../../src/middleware/cf-operation/types";
+
+import { NetworkContext } from "../../../src/utils/network-context";
+import { Signature } from "../../../src/utils/signature";
+
+describe("CfOpSetState", () => {
+  const TEST_NETWORK_CONTEXT = new NetworkContext(
+    ethers.utils.hexlify(ethers.utils.randomBytes(20)),
+    ethers.utils.hexlify(ethers.utils.randomBytes(20)),
+    ethers.utils.hexlify(ethers.utils.randomBytes(20)),
+    ethers.utils.hexlify(ethers.utils.randomBytes(20)),
+    ethers.utils.hexlify(ethers.utils.randomBytes(20)),
+    ethers.utils.hexlify(ethers.utils.randomBytes(20)),
+    ethers.utils.hexlify(ethers.utils.randomBytes(20)),
+    ethers.utils.hexlify(ethers.utils.randomBytes(20))
```

Comment:
It's better not to use random input for your tests unless you're explicitly testing what your code does under random inputs. I would recommend using "0x11111.." etc instead